File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n= nums.size();
        int i1; int i2;
        unordered_map <int, int> m;
        for (int i=0; i<n; i++){
            int rem= target-nums[i];
            if (m.find(rem)!=m.end()){
                i1=i;
                i2= m[rem];
                break;
            }
            m[nums[i]]= i;
        }
        vector <int> ans= {i1, i2};
        return ans;
    }
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        ans=[]

        n= len(nums)
        hash_map= {}
        for i in range (n):
            rem=target-nums[i]
            if rem in hash_map:
                ans.append(i)
                ans.append(hash_map[rem])
                return ans

            hash_map[nums[i]] = i

            

        